### PR TITLE
Added GuiRegistry

### PR DESCRIPTION
**Pull request by laci200270**

### CODE REVIEW

**Comment by heldplayer** [comment]
Besides the complete lack of any signs of formatting, what use would this GuiRegistry serve? What is the motivation to having this in?

**Comment by laci200270** [comment]
@heldplayer With this you don't need to use numbers for identityfing GUIs in your mod simple use strings.And also I'll write a json based GUI System on top of this

File: src/main/java/net/minecraftforge/common/gui/GuiContainerPair.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+        public final Container container;
+        public final Gui gui;
+
+        public GuiContainerPair(Container container, Gui gui){
+            this.container=container;
+            this.gui=gui;
+        }
+
+    }
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/common/gui/GuiContainerPair.java]: So the entire point of this class is to hard tie a client and server classes together therefor killing ANY chance the dedicated server works on it's own?

File: src/main/java/net/minecraftforge/common/gui/TileEntityGuiRegistryHandler.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Override
+    public GuiContainerPair handle(Object mod, String id, TileEntity tileEntity, EntityPlayer player) {
+        GuiRegistry.GuiHandlerMetadata gconpair= getGuiRegistry().getGuiRegistry(mod).datas.get(id);
+        try {
+            Container container=null;
+            Gui gui=null;
+            container= (Container) gconpair.container.getConstructor(new Class[]{TileEntity.class, EntityPlayer.class}).newInstance(tileEntity,player);
+            gui= (Gui) gconpair.gui.getConstructor(new Class[]{Container.class}).newInstance(container);
+            GuiContainerPair pair;
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/common/gui/TileEntityGuiRegistryHandler.java]: No, Just no.

**Comment by LexManos** [comment]
You know what, rather then just tearing this apart for being stupid.
Your solution is to completely break the client and server disconnect all because you don't know how to do "SomeStringID".hashCode() ?

**Pull request closed without merging**